umbral-pre-python/umbral_pre/__init__.pyi

```diff
@@ -201,35 +208,34 @@ class VerifiedCapsuleFrag:
     def to_bytes_simple(self) -> bytes:
         ...
 
-
-def reencrypt(capsule: Capsule, kfrag: VerifiedKeyFrag) -> VerifiedCapsuleFrag:
-    ...
-
-
 def decrypt_reencrypted(
         receiving_sk: SecretKey,
         delegating_pk: PublicKey,
         capsule: Capsule,
-        cfrags: Sequence[VerifiedCapsuleFrag],
+        verified_cfrags: Sequence[VerifiedCapsuleFrag],
         ciphertext: bytes,
-        ) -> Optional[bytes]:
+) -> Optional[bytes]:
     ...
 
+def reencrypt(capsule: Capsule, kfrag: VerifiedKeyFrag) -> VerifiedCapsuleFrag:
+    ...
 
-class CurvePoint:
 
-    def coordinates(self) -> Tuple[bytes, bytes]:
-        ...
+@final
+class CurvePoint:
+    coordinates: Tuple[bytes, bytes]
 
 
+@final
 class Parameters:
 
-    def __init__(self) -> None:
+    def __init__(self, *args) -> None:
```

Comment:
This doesn't seem right, it doesn't take any arguments.

Reply:
Perhaps it's a false positive by `mypy.stubtest`.

But since it's a constructor that doesn't take any parameters, shouldn't it be [`__new__`](https://pyo3.rs/v0.18.2/class.html#constructor)?

Reply:
Are you referring to the Python's `__new__`, or to the `#[new]` annotation? The latter is set in the bindings.

Reply:
I was referring to the `__new__`. In `pyo3` it's not possible to override `__init__` method, you may only use `#[new]` to override `__new__`.

The runtime signatures generated by `pyo3` look like this:
```
>>> inspect.signature(Parameters.__init__)
<Signature (self, /, *args, **kwargs)>
>>> inspect.signature(Parameters.__new__)
<Signature (*args, **kwargs)>
```
So if we didn't annotate `__init__` in `.pyi` stub, we would end up having these signatures. But if we do, `mypy.stubtest` is complaining about the mismatch between `.pyi` and runtime signatures (because there is one):

```
error: umbral_pre.Parameters.__init__ is inconsistent, stub does not have *args argument "args"
Stub: in file ../rust-umbral/umbral-pre-python/umbral_pre/__init__.pyi:237
def (self: umbral_pre.Parameters)
Runtime:
def (self, /, *args, **kwargs)
```
As I see it, the options are:
1. Remove `__init__` from `.pyi`
2. Add `def __init__(self, *args) -> None:` to `.pyi`
3. Disable this error check in `mypy.stubtest`

I will go ahead and implement 3.